### PR TITLE
Filter assignments by course

### LINKED ISSUE #145: Filter assignments by course

### PR DESCRIPTION
**Pull request by JonasWanke**
**Please merge #256 first!**
**Closes: #145**

`AssignmentsScreen` can now be filtered by course. This also includes quick-filtering by tapping the course-chip on an `AssignmentCard` and parsing the query string from incoming deep links.

**Known issues:**
- I didn't get `DraggableScrollableSheet` to work inside a `ModalBottomSheet`, so there is an overflow if the SortFilter sheet is higher than the screen.

**Screenshots**

| SortFilter sheet | Active course filter  |
| :------------: | :------------: |
| image | image |

### CODE REVIEW

File: lib/app/sort_filter/filtering.dart [hunk 1]
```diff
@@ -149,7 +154,87 @@ class DateRangeFilterSelection {
...
+        return ChipGroup(
+          children: <Widget>[
+            for (final category in categories)
+              FilterChip(
+                selected: selection.contains(category),
+                onSelected: (isSelected) {
+                  if (isSelected) {
+                    updater({...selection, category});
+                  } else {
+                    updater(selection.whereNot((c) => c == category).toSet());
```
Comment (MarcelGarus) [inline lib/app/sort_filter/filtering.dart]:
```suggestion
                    updater(selection.toSet().difference({category}));
```
or
```suggestion
                    updater(selection.toSet()..remove(category));
```

File: lib/assignment/widgets/assignments_screen.dart [hunk 2]
```diff
@@ -132,24 +133,28 @@ class _AssignmentsScreenState extends State<AssignmentsScreen>
   }
 }
 
+typedef OnCourseClicked = void Function(Id<Course> courseId);
```
Comment (MarcelGarus) [inline lib/assignment/widgets/assignments_screen.dart]: Hmm. It doesn't really sound like a type. What about `CourseClickedCallback`?
Also, alliterations are fun.

**Review by MarcelGarus — CHANGES_REQUESTED** [review]

**Comment by janrenz** [comment]
What if the list of courses gets too long?

**Comment by JonasWanke** [comment]
@janrenz The intended effect would be a scrollable sheet (probably using `DraggableScrollableSheet`), but I didn't get that to work yet (i.e., it currently overflows the screen). We will still merge this PR and look into scrolling soon (#260).

**Review by MarcelGarus — APPROVED** [review]

**Pull request merged**